### ISSUE TITLE
Crash using Fire Storm in Slime

### ISSUE THREAD

**Issue by jamesunger** [crawl/crawl#2241]
The crash triggers after casting Fire Storm in Slime. It seems like it is related to the spawned fire vortexes after words and/or noise and slime walls, but I'm not really sure. It is easy to trigger by just blasting Fire Storm a few times and waiting for the vortexes to move around.

Platform: Ubuntu Linux tiles (but also crashes on local web tiles and console debug)
Version info: 0.28-a0-906-g0406aca7d3

[taisin3.txt](https://github.com/crawl/crawl/files/7592536/taisin3.txt) - morgue
[crash-taisin3-20211123-222018.txt](https://github.com/crawl/crawl/files/7592542/crash-taisin3-20211123-222018.txt) - crash file with debug
[taisin3.cs.gz](https://github.com/crawl/crawl/files/7592546/taisin3.cs.gz) - save file

**Comment by jamesunger** [comment]
Terrible hack, but prevents the crashes.

```diff --git a/crawl-ref/source/mon-behv.cc b/crawl-ref/source/mon-behv.cc
index dc93792932..cdf55e9c59 100644
--- a/crawl-ref/source/mon-behv.cc
+++ b/crawl-ref/source/mon-behv.cc
@@ -1028,6 +1028,9 @@ void behaviour_event(monster* mon, mon_event_type event, const actor *src,
             if (mon->is_patrolling())
                 break;

+           if (mon->type == MONS_FIRE_VORTEX)
+               break;
+
             ASSERT(!src_pos.origin());
             mon->target = src_pos;
         }
```

**Comment by ebering** [comment]
Fixed in 59e3c9d
